funsor/distribution.py

```diff
@@ -447,3 +447,24 @@ def eager_mvn(loc, scale_tril, value):
     inputs[var] = reals(scale_diag.shape[0])
     gaussian = log_prob + Gaussian(info_vec, precision, inputs)
     return gaussian(**{var: value - loc})
+
+
+def eager_dirichlet_multinomial(red_op, bin_op, reduced_vars, x, y):
```

Comment:
Can you move this to `torch/distributions.py` since that's the only backend with `DirichletMultiNomial`?